src/GraphQL/Language/AST/Fields.cs

```diff
@@ -17,22 +24,31 @@ public class Fields : Dictionary<string, Field>
         /// <br/><br/>
         /// See http://spec.graphql.org/June2018/#sec-Field-Collection and http://spec.graphql.org/June2018/#CollectFields()
         /// </summary>
-        public Fields CollectFrom(ExecutionContext context, IGraphType specificType, SelectionSet selectionSet)
+        /// <param name="context"></param>
+        /// <param name="specificType"></param>
+        /// <param name="selectionSet"></param>
+        /// <param name="fieldCollectionStrategy">
+        /// If set to <see langword="null"/>, <see cref="ExecutionHelper.ShouldIncludeNode"/> will be used to work
+        /// as required by the specification. Set this parameter if you understand exactly what you are doing,
+        /// because your actions may lead to the fact that the server's behavior ceases to comply with the
+        /// specification requirements.
+        /// </param>
+        public Fields CollectFrom(ExecutionContext context, IGraphType specificType, SelectionSet selectionSet, IFieldCollectionStrategy fieldCollectionStrategy = null)
         {
             List<string> visitedFragmentNames = null;
-            CollectFields(context, specificType, selectionSet, ref visitedFragmentNames);
+            CollectFields(context, specificType, selectionSet, fieldCollectionStrategy ?? DefaultFieldCollectionStrategy.Instance, ref visitedFragmentNames);
```

Comment:
Why not store the field collection strategy instance in the context and/or use the execution strategy which is stored there? Sorry I’m not at a pc just now so maybe it’s obvious. I guess at first glance I wouldn’t add this argument.

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/pull/2294/files#diff-43fbf64bffe7f823cf1060bfa739a3ec85efdd804dd414e11fadef48cbb04c82

Reply:
Ooops , understood.

Reply:
I got lost in my own changes. Done!